src/math/trig.c

```diff
@@ -94,9 +97,197 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine, cosine*/
 	*th_cdc_fxp = th_rad_fxp;
+}
+
+/**
+ * \CORDIC-based approximation for inverse cosine
+ * Arguments	: int32_t cosvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE]
+ * Return Type	: int32_t
+ */
+int32_t is_scalar_cordic_acos(int32_t cosvalue, int16_t numiters,
+			      const int32_t *LUT)
+{
+	int32_t b_cosvalue;
+	int32_t b_i;
+	int32_t i;
+	int32_t x = 0;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y = 0;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z = 0;
+	int32_t sign;
+	int16_t j;
+	int16_t k;
 
+	b_cosvalue = (cosvalue >> 1);
+	/* Initialize the variables for the cordic iteration
+	 * angles less than pi/4, we initialize (x,y) along the x-axis.
+	 * angles greater than or equal to pi/4, we initialize (x,y)
+	 * along the y-axis. This improves the accuracy of the algorithm
+	 * near the edge of the domain of convergence
+	 */
+	if (b_cosvalue < cord_arcsincos_q28fl) {
+		x = 0;
+		y = cord_arcsincos_q30fl;
+		z = PI_DIV2_Q3_29;
+	} else {
+		x = cord_arcsincos_q30fl;
+		y = 0;
+		z = 0;
+	}
+
+	b_cosvalue <<= 1;
```

Comment:
You have right-shifted b_cosvalue by 1 before for comparison, and now left-shift it back. It will lose a slight precision, and look unnecessary.

How about just set b_cosvalue = cosvalue here, and rephrase line#135 as
if (cosvalue < (cord_arcsincos_q28fl << 1))
or
if ((cosvalue >> 1) < cord_arcsincos_q28fl)

Reply:
Yup, we can save `b_cosvalue` space

Reply:
With       [w]      `if ((cosvalue >> 1) < cord_arcsincos_q28fl)`
WithOut [w/o]  `if (b_cosvalue < cord_arcsincos_q28fl)`

Please see the impact on few sampled output  
```
+--------+---------------+----------------+-------------+---------------+
|idx	 |ipfxp[-1:0.1:1]|32B_opfxp(w/o)  |32B_opfxp(w) |    comment   	|
+--------+---------------+----------------+-------------+---------------+
|[0]	 |-1073741824	 |1686629713      |1686629713 	|               |
|[1]	 |-966367642	 |1444486531      |1444486531 	|               |
|[2]	 |-858993459	 |1341152679      |1341152679 	|               |
|[3]	 |-751619277	 |1259603211      |1259603213 	|	x       |
|[4]	 |-644245094	 |1188791875      |1188791875 	|               |
|[5]	 |-536870912	 |1124419797      |1124419797 	|               |
|[6]	 |-429496730	 |1064246279      |1064246279 	|               |
|[7]	 |-322122547	 |1006895465      |1006895465 	|               |
|[8]	 |-214748365	 |951418065	  |951418067  	|	x       |
|[9]	 |-107374182	 |897091831	  |897091831  	|               |
|[10]	 |0         	 |843314856	  |843314856  	|               |
|[11]	 |107374182	 |789537882	  |789537882  	|               |
|[12]	 |214748365	 |735211648	  |735211646  	|	x       |
|[13]	 |322122547	 |679734248	  |679734248  	|               |
|[14]	 |429496730	 |622383434	  |622383434  	|               |
|[15]	 |536870912	 |562209916	  |562209916  	|               |
|[16]	 |644245094	 |497837838	  |497837838  	|               |
|[17]	 |751619277	 |427026502	  |427026500  	|	x       |
|[18]	 |858993459	 |345477034	  |345477034  	|               |
|[19]	 |966367642	 |242143182	  |242143182  	|               |
|[20]	 |1073741824 	 |0		  |0          	|               |

```